account_invoice_constraint_chronology/tests/test_account_constraint_chronology.py

```diff
@@ -94,23 +66,26 @@ def test_invoice_draft(self):
         create_simple_invoice(self, journal.id, date)
         date = today.strftime(DEFAULT_SERVER_DATE_FORMAT)
         invoice_2 = create_simple_invoice(self, journal.id, date)
-        self.assertRaises(exceptions.Warning, workflow.trg_validate, self.uid,
-                          'account.invoice', invoice_2.id, 'invoice_open',
-                          self.cr)
+        self.assertTrue((invoice_2.state == 'draft'),
+                        "Initially bill state is not Draft")
+        with self.assertRaises(exceptions.Warning):
+            invoice_2.action_invoice_open()
 
     def test_invoice_validate(self):
         journal = get_journal_check(self, True)
         today = datetime.now()
         tomorrow = today + timedelta(days=1)
         date = tomorrow.strftime(DEFAULT_SERVER_DATE_FORMAT)
-        invoice = create_simple_invoice(self, journal.id, date)
-        workflow.trg_validate(self.uid, 'account.invoice', invoice.id,
-                              'invoice_open', self.cr)
+        invoice_1 = create_simple_invoice(self, journal.id, date)
+        self.assertTrue((invoice_1.state == 'draft'),
+                        "Initially bill state is not Draft")
         date = today.strftime(DEFAULT_SERVER_DATE_FORMAT)
         invoice_2 = create_simple_invoice(self, journal.id, date)
-        self.assertRaises(exceptions.Warning, workflow.trg_validate, self.uid,
-                          'account.invoice', invoice_2.id, 'invoice_open',
-                          self.cr)
+        self.assertTrue((invoice_2.state == 'draft'),
+                        "Initially bill state is not Draft")
+        with self.assertRaises(exceptions.Warning):
+            invoice_1.action_invoice_open()
+            invoice_2.action_invoice_open()
```

Comment:
`invoice_1.action_invoice_open()` must not raise, if I read previous version correctly, so move it out of with self.assertRaises.